### PR TITLE
support half float in data inspector

### PR DESCRIPTION
**Pull request by zebe**
Like it says on the tin, this lets you select 2 bytes and have them be interpreted as a half float type.

### CODE REVIEW

File: app/sources/DataInspector.m [hunk 1]
```diff
@@ -249,6 +249,17 @@ static long double ieeeToLD(const void *bytes, unsigned exp, unsigned man) {
...
+        case sizeof(short):
+        {
+            union {
+                uint16_t i;
+                __fp16 f;
+            } temp;
+            _Static_assert(sizeof temp.f == sizeof temp.i, "sizeof(uint16_t) is not 2!");
+            temp.i = *(const uint16_t *)bytes;
+            if (endianness != eNativeEndianness) temp.i = (uint16_t)reverse(temp.i, sizeof(uint16_t));
+            return [NSString stringWithFormat:@"%.15g", temp.f];
```
Comment (kainjow) [inline app/sources/DataInspector.m]: I think this needs casting to `double`. In a test project if I do the same I get:

> Implicit conversion increases floating-point precision: '__fp16' to 'double'

File: app/sources/DataInspector.m [hunk 2]
```diff
@@ -249,6 +249,17 @@ static long double ieeeToLD(const void *bytes, unsigned exp, unsigned man) {
 
 static id floatingPointDescription(const unsigned char *bytes, NSUInteger length, enum Endianness_t endianness) {
     switch (length) {
+        case sizeof(short):
```
Comment (kainjow) [inline app/sources/DataInspector.m]: I think `short` here should be `uint16_t`.

I'll update the others too so it's not mixing types.

**Comment by kainjow** [comment]
Otherwise looks good. I didn't even know 2-byte float existed :)

**Comment by zebe** [comment]
Thanks. Yeah, it's not all that common since it's a pretty small space, but it gets used in graphics a fair bit. Should I go make those changes on my branch?

**Comment by kainjow** [comment]
Yes please.

**Comment by zebe** [comment]
OK, I think that addresses both comments.

**Comment by kainjow** [comment]
Thanks!

**Comment by zebe** [comment]
Cheers!

**Pull request merged**